src/qibocal/protocols/state_tomography.py

```diff
@@ -140,30 +143,28 @@ def _fit(data: StateTomographyData) -> StateTomographyResults:
     target_density_matrix_real = {}
     target_density_matrix_imag = {}
     fid = {}
-    for qubit in data.qubits:
-        x_exp, y_exp, z_exp = (
-            1 - 2 * np.mean(data[qubit, basis].samples) for basis in BASIS
+    circuit = data.circuit
+    circuit.density_matrix = True
+    total_density_matrix = NumpyBackend().execute_circuit(circuit=circuit).state()
+    for i, qubit in enumerate(data.targets):
+        traced_qubits = list(range(len(data.qubits)))
+        traced_qubits.remove(i)
```

Comment:
```suggestion
        traced_qubits = [q for q in range(len(data.qubits)) if q != i]
```
slightly more readable...